### PR TITLE
Preliminary imageToGround code using ray intersections

### PR DESCRIPTION
**Pull request by MarjorieH**
Intersection code has been implemented, taken from sensorModelRefactor. Still needs correct look direction in order to function properly. Also needs distortion model to produce accurate results.
DO NOT MERGE :warning:

### CODE REVIEW

**Comment by jlaura** [comment]
Can one of the admins verify this patch?

**Comment by ihumphrey-usgs** [comment]
:+1: :fireworks: :1st_place_medal:

File: src/objs/MdisNacSensorModel.cpp [hunk 1]
```diff
@@ -6,15 +6,37 @@
 const std::string MdisNacSensorModel::_SENSOR_MODEL_NAME 
                                       = "ISIS_MDISNAC_USGSAstro_1_Linux64_csm30.so";
 
+                                      
 MdisNacSensorModel::MdisNacSensorModel() {
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: Please add comments describing the purpose of these variables.

File: src/objs/MdisNacSensorModel.cpp [hunk 2]
```diff
@@ -35,16 +58,253 @@ csm::ImageCoordCovar MdisNacSensorModel::groundToImage(const csm::EcefCoordCovar
...
+                                                 csm::WarningList *warnings) const {
+
+  double sample = imagePt.samp;
+  double line = imagePt.line;
+
+  // center the sample line
+  sample -= m_ccdCenter - 0.5; // ISD needs a center sample in CSM coord
+  line -= m_ccdCenter - 0.5; // ISD needs a center line in CSM coord (.5 .5 pixel centers)
+  
+  // Convert from sample/line to focal plane coordinates
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: $$\sin(x)$$

File: src/objs/MdisNacSensorModel.cpp [hunk 3]
```diff
@@ -35,16 +58,253 @@ csm::ImageCoordCovar MdisNacSensorModel::groundToImage(const csm::EcefCoordCovar
...
+  
+  // Trigonometric functions for rotation matrix
+  const double sinw = std::sin(m_omega);
+  const double sinp = std::sin(m_phi);
+  const double sink = std::sin(m_kappa);
+  const double cosw = std::cos(m_omega);
+  const double cosp = std::cos(m_phi);
+  const double cosk = std::cos(m_kappa);
+
+  // Rotation matrix taken from Introduction to Photogrammetry by Edward M. Mikhail
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: _Introduction to Modern Photogrammetry_ by Edward M. Mikhail, et al., p. 373

File: src/objs/MdisNacSensorModel.cpp [hunk 4]
```diff
@@ -35,16 +58,254 @@ csm::ImageCoordCovar MdisNacSensorModel::groundToImage(const csm::EcefCoordCovar
...
+}
+
+
+std::vector<double> MdisNacSensorModel::perpendicular(const std::vector<double> &v1, 
+                                                      const std::vector<double> &v2) const {
+  
+  double max1 = 0.0;
+  double max2 = 0.0;
+  
+  // Get the infinite norm for each vector.
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: The infinity norm is the maximum value in a vector, not the absolute value.

File: src/objs/MdisNacSensorModel.cpp (hunk 1 above)
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: Add function descriptions to all your functions (especially for the geometric ones such as project).  ASCII art would be awesome!

File: src/objs/MdisNacSensorModel.cpp [hunk 5]
```diff
@@ -35,16 +58,254 @@ csm::ImageCoordCovar MdisNacSensorModel::groundToImage(const csm::EcefCoordCovar
...
+  
+  newV2[0] = v2[0] / max2;
+  newV2[1] = v2[1] / max2;
+  newV2[2] = v2[2] / max2;
+  
+  // Project first vector onto second vector.
+  std::vector<double> parallelV = project(newV1, newV2);
+  
+  std::vector<double> perpendicularV(3);
+
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: Gram-Schmidt process

File: src/objs/MdisNacSensorModel.cpp [hunk 6]
```diff
@@ -35,16 +58,254 @@ csm::ImageCoordCovar MdisNacSensorModel::groundToImage(const csm::EcefCoordCovar
...
+  // onto the second.
+  perpendicularV[0] = v1[0] - parallelV[0] * max1;
+  perpendicularV[1] = v1[1] - parallelV[1] * max1;
+  perpendicularV[2] = v1[2] - parallelV[2] * max1;
+  
+  return perpendicularV;
+}
+
+
+std::vector<double> MdisNacSensorModel::project(const std::vector<double> &v1,
```
Comment (twilson271828) [inline src/objs/MdisNacSensorModel.cpp]: Add commentary here (and ASCII art would be good, showing what is being projected onto what).

**Pull request merged**